core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -263,7 +261,7 @@ public void preload() {
       long start = timeService.time();
 
 
-      final long maxEntries = getMaxEntries();
+      final long maxEntries = configuration.memory().size();
```

Comment:
Hrmm this defaults to -1, should we not set this to `Long.MAX_VALUE` or ignore if it is -1?